internal/engine/eval/vulncheck/review.go

```diff
@@ -260,32 +222,35 @@ func (ra *reviewPrHandler) submit(ctx context.Context) error {
 		return fmt.Errorf("could not find previous review: %w", err)
 	}
 
-	if ra.minderReview != nil {
-		if ra.minderReview.CommitID != nil && *ra.minderReview.CommitID == ra.pr.CommitSha {
-			// if the previous review was on the same commit, keep it
-			ra.logger.Debug().
-				Int64("review-id", ra.minderReview.GetID()).
-				Msg("previous review was on the same commit, will keep it")
-			return nil
-		}
-
-		err := ra.dismissReview(ctx)
-		if err != nil {
-			ra.logger.Error().Err(err).
-				Int64("review-id", ra.minderReview.GetID()).
-				Msg("could not dismiss previous review")
-		}
+	// if the previous review was on the same commit, keep it
+	if ra.minderReview != nil && ra.minderReview.GetCommitID() == ra.pr.GetCommitSha() {
 		ra.logger.Debug().
 			Int64("review-id", ra.minderReview.GetID()).
-			Msg("dismissed previous review")
+			Msg("previous review was on the same commit, will keep it")
+		return nil
+	}
+
+	if err := ra.dismissReview(ctx); err != nil {
+		return fmt.Errorf("could not dismiss previous review: %w", err)
```

Comment:
Note that previously error on dismiss review was not fatal, this was on purpose so that minder wouldn't error out trying to dismiss its own review running as the same use. I wonder if we could add this to body of `dismissReview`:
```
+       if ra.pr.AuthorId == ra.authorizedUser.GetID() {
+               ra.logger.Debug().Msg("author is the same as the authenticated user, can't dismiss")
+               return nil
+       }
+
```
this is pretty much the same as ignoring the error, except we don't actually attempt the call so we save a token..